### PR TITLE
Improve performance when using exclusion criteria on large data sets.

### PR DESCRIPTION
**Pull request by gijskant**

### CODE REVIEW

File: src/app/services/transmart-services/transmart-resource.service.ts [hunk 1]
```diff
@@ -255,24 +271,24 @@ export class TransmartResourceService {
...
     return new Promise<any>((resolve, reject) => {
       if (this.autosaveSubjectSets) {
-        if (totalCountItem) {
-          this.inclusionCounts.patientCount = totalCountItem.patientCount + this.exclusionCounts.patientCount;
-          this.inclusionCounts.observationCount = totalCountItem.observationCount + this.exclusionCounts.observationCount;
-          resolve(true);
+        if (!totalCountItem) {
+          return reject('Missing transmart total-count item to calculate inclusion counts.');
+        }
+        this.inclusionCounts.patientCount = totalCountItem.patientCount - this.exclusionCounts.patientCount;
```
Comment (ewelinagr) [inline src/app/services/transmart-services/transmart-resource.service.ts]: hmm, are you sure about the subtraction character? Previously `total = inclusion - exclusion`, so `inclusion = total + exclusion` (there was a `+` here).

Comment (gijskant) [reply src/app/services/transmart-services/transmart-resource.service.ts]: You are correct, I will fix it.

**Comment by gijskant** [comment]
Still some weird issues with 'step 2' when using `autosave-subject-sets`.

**Comment by ewelinagr** [comment]
Looks good! Though the rabasing of fractalis-integration branch with dev might be more challenging now :)

**Pull request merged**